src/Uno.Extensions.Templates/content/unoapp-extensions/MyExtensionsApp.UI/HostConfiguration.cs

```diff
@@ -0,0 +1,138 @@
+				})
+		);
+	}
+
+	public static IApplicationBuilder UseEnvironment(this IApplicationBuilder builder)
+	{
+#if DEBUG
+		// Switch to Development environment when running in DEBUG
+		builder.Configure(host => host.UseEnvironment(Environments.Development));
+#endif
+		return builder;
+	}
+
+	public static IApplicationBuilder UseLogging(this IApplicationBuilder builder)
+	{
+		// Add platform specific log providers
+		return builder.Configure(host => host.UseLogging(configure: (context, logBuilder) =>
+		{
+			// Configure log levels for different categories of logging
+			logBuilder.SetMinimumLevel(
+				context.HostingEnvironment.IsDevelopment() ?
+					LogLevel.Information :
+					LogLevel.Warning);
+		}, enableUnoLogging: true));
+	}
+
+	public static IApplicationBuilder UseConfiguration(this IApplicationBuilder builder)
+	{
+		return builder.Configure(host => host.UseConfiguration(configure: configBuilder =>
+			configBuilder
+				.EmbeddedSource<App>()
+				.Section<AppConfig>()
+		));
+	}
+
+	public static IApplicationBuilder UseLocalization(this IApplicationBuilder builder)
+	{
+		// Enable localization (see appsettings.json for supported languages)
+		return builder.Configure(host => host.UseLocalization());
+	}
+
+	public static IApplicationBuilder UseSerialization(this IApplicationBuilder builder)
+	{
+		// Register Json serializers (ISerializer and ISerializer)
+		return builder.Configure(host => host.UseSerialization());
+	}
+
+	public static IApplicationBuilder ConfigureServices(this IApplicationBuilder builder)
+	{
+		// Register services for the application
+		return builder.Configure(host => host.ConfigureServices(ConfigureServices));
+	}
+
+	public static IApplicationBuilder UseNavigation(this IApplicationBuilder builder)
+	{
+		// Enable navigation, including registering views and viewmodels
+		return builder.Configure(host => host.UseNavigation(
+//+:cnd:noEmit
+#if(reactive)
+			ReactiveViewModelMappings.ViewModelMappings,
+#endif
+//-:cnd:noEmit
+			RegisterRoutes)
+
+			// Add navigation support for toolkit controls such as TabBar and NavigationView
+			.UseToolkitNavigation());
+	}
+
+	public static async Task<IHost> ShowAsync<TShell>(this IApplicationBuilder appBuilder)
+		where TShell : UIElement, new()
+	{
+		var appRoot = new TShell();
+		var navRoot = appRoot as ContentControl;
+		if (appRoot is IContentControlProvider contentProvider)
+		{
+			navRoot = contentProvider.ContentControl;
+		}
+
+		if (navRoot is ExtendedSplashScreen splashScreen)
+		{
+			splashScreen.Initialize(appBuilder.Window, appBuilder.Arguments);
+		}
+
+		appBuilder.Window.Content = appRoot;
+		appBuilder.Window.Activate();
+
+		// TODO: This needs tidying up inside Navigation library so that there's a single
+		// InitializeNavigationAsync method that can make the determination on whether
+		// to use the logic from Navigation.Toolkit or not
+		if (navRoot is LoadingView loading)
```

Comment:
Two things that need resolving so that we can move ShowAsync into extensions library:
- we need a single InitializeNavigationAsync method that can make the determination on whether to use the logic from Navigation.Toolkit or not
- we need a way that the InitializeNavigationAsync parameter can be specified - allows for dynamic determination of first navigation request (for example if user is logged in go to home page, otherwise login page)